proxy/src/lib.rs

```diff
@@ -1,4 +1,109 @@
-#![deny(clippy::undocumented_unsafe_blocks)]
+// rustc lints/lint groups
+// https://doc.rust-lang.org/rustc/lints/groups.html
+#![deny(
+    deprecated,
+    future_incompatible,
+    // TODO: consider let_underscore
+    nonstandard_style,
+    rust_2018_idioms,
+    rust_2021_compatibility,
+    rust_2024_compatibility
+)]
+#![warn(clippy::all, clippy::pedantic, clippy::nursery, clippy::cargo)]
```

Comment:
```suggestion
#![warn(clippy::all, clippy::pedantic, clippy::cargo)]
```

I don't mind pedantic too much, but nursery is probably too inconsistent.

Reply:
Done.

I also removed the exceptions.